src/@types/rateLimit.d.ts

```diff
@@ -23,6 +23,11 @@ export interface RedisBucket {
     timestamp: number;
 }
 
+export interface RedisWindow {
```

Comment:
You created interface `RedisWindow` which looks the same as the interface `RedisBucket`. Any reason you can't use `RedisBucket`?

Reply:
going to add more values in the future to track the rolling and fixed window timestamps

Reply:
In that case the new type should extend `RedisBucket` to keep things DRY. 

At a higher level, what data does Redis need to track? My understanding is the complexity of all requests in the previous window along with the complexity of all requests in the current window. In each case the timestamp could just represent the beginning of the window. I don't think we need to track the timestamp of each request in Redis. 

If we don't need anything else then for each user we just need to store two `RedisBuckets` either under to separate keys or both as an array. In the latter case then we would just need to type something as `[RedisBucket, RedisBucket]` 

Thoughts?

Reply:
Discussed offline